pkgs/development/libraries/pipewire/wireplumber.nix

```diff
@@ -0,0 +1,53 @@
+{ pkgs, lib, stdenv, fetchFromGitLab }:
+stdenv.mkDerivation rec {
+  pname = "wireplumber";
+  version = "0.4.5";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "pipewire";
+    repo = "wireplumber";
+    rev = version;
+    sha256 = "sha256-EQw1hVQ0rsB9CWOBtcP1enpIwkHmNTr3AJm3rFiJpsw=";
+  };
+
+  mesonFlags = [
+    "-Dsystem-lua=true"
+    "-Delogind=disabled"
+  ];
+
+  nativeBuildInputs = with pkgs; [
+    meson
+    pkg-config
+    ninja
+
+    # for docs
+    (python39.withPackages (ps: with ps; [
+      sphinx
+      sphinx_rtd_theme
+      breathe
+      lxml
+    ]))
+    doxygen
+    graphviz
+
+    gobject-introspection
```

Comment:
This likely needs to be optional on `stdenv.hostPlatform == stdenv.buildPlatform` due to [longstanding issues wrt cross-compiling gobject-introspection](https://github.com/NixOS/nixpkgs/pull/88222).

Reply:
Added a separate `enableGI` argument that defaults to this, if someone really wants to shoot themselves in the foot, they can go ahead and override.